### DISCUSSION TITLE
Communities join screens (Non token gated) - Add owner node offline information box

### DISCUSSION THREAD

**Discussion by J-Son89** [status-im/status-legacy#16658] [General]
We can't actually check when the owner is online, instead we will only show this "owner not online" box whenever there is a pending request. With the intention of optimising this approach at a later point.

Add notice to user that the user node is offline as per the designs.

https://www.figma.com/file/h9wo4GipgZURbqqr1vShFN/Communities-for-Mobile?node-id=7035%3A462899&t=ivg9IboexBv4dlSR-0

add integration tests to this.

**Comment by vkjr** [comment]
@J-Son89, just to clarify, do I correctly understand that we show "Community owner node is offline" only in case when the community is free to join (no approve required) and our request is till pending)?

**Comment by J-Son89** [comment]
@VladimrLitvinenko I think it's good to get this clarification in public and also helps alongside the context of the Github issue, but can you raise this question in Mobile-ui so that the design team can see it too.  They don't really use Github.

From what I understand we just want to show this message to a user at the same time their request is pending. 

So this can be in all 4 cases. i.e this can be in both token-gated and non-token gated communities and when manual approval is needed and not needed.

**Comment by vkjr** [comment]
@J-Son89, I believe that in cases when a user request should be accepted by a community owner, this message would be both incorrect and redundant. Incorrect - because maybe community owner IS online, but simply don't want you in the community) Redundant - because there is "Pending" badge. Wdyt?

**Comment by J-Son89** [comment]
> @J-Son89, I believe that in cases when a user request should be accepted by a community owner, this message would be both incorrect and redundant. Incorrect - because maybe community owner IS online, but simply don't want you in the community) Redundant - because there is "Pending" badge. Wdyt?

let's have this conversation on Status - mobile ui channel so design team is brought into the conversation :)

**Comment by vkjr** [comment]
> @J-Son89,  
> let's have this conversation on Status - mobile ui channel so design team is brought into the conversation :)

Asked there)

**Comment by du82** [comment]
this can be closed as the info box was added in #14515

**Comment by vkjr** [comment]
Not yet, unfortunately)
The current implementation in #14515 adds the info box, but there is an unfinished discussion about when it should show up. 

In the Figma designs, info box shows up only for communities that are free to join and only in cases when the community owner node is offline. 
But since we don't really have information about when a community node is online/offline, the question was asked to the design team, and decided to wait for the decision of Join (product owner). Unfortunately, I don't know his github nickname to tag him here (@cammellos, maybe you know?)

So current implementation shows infobox when: the community is open to join + user issued a join request + user didn't yet join the community. But here is the problem.. in our code community gets `:joined true` key even if you tried to join when network was disabled. I don't know if it is correct behavior since not really familiar with the machinery behind joining the open communities. Can we really consider ourselves "joined" immediately or do we expect some response from another node that may fail? @cammellos, maybe you can share some details as the one with extensive backend knowledge?

**Comment by cammellos** [comment]
> Not yet, unfortunately) The current implementation in #14515 adds the info box, but there is an unfinished discussion about when it should show up.
> 
> In the Figma designs, info box shows up only for communities that are free to join and only in cases when the community owner node is offline. But since we don't really have information about when a community node is online/offline, the question was asked to the design team, and decided to wait for the decision of Join (product owner). Unfortunately, I don't know his github nickname to tag him here (@cammellos, maybe you know?)
> 
> So current implementation shows infobox when: the community is open to join + user issued a join request + user didn't yet join the community. But here is the problem.. in our code community gets `:joined true` key even if you tried to join when network was disabled. I don't know if it is correct behavior since not really familiar with the machinery behind joining the open communities. Can we really consider ourselves "joined" immediately or do we expect some response from another node that may fail? @cammellos, maybe you can share some details as the one with extensive backend knowledge?

@vkjr 

The `joined` issue could be a relic of the old implementation, looks like it hasn't been changed completely, do you see yourself in the member list if the admin accepts the request?

**Comment by vkjr** [comment]
This issue relates to communities that do not require admin approval

**Comment by cammellos** [comment]
> This issue relates to communities that do not require admin approval

Both of them requires approval looking from the comments on the code https://github.com/status-im/status-go/blob/dd8be1d8b041f967a0707cfb70ac96c5a71d93a8/protocol/communities/community.go#L1205 , it might be that we are not sending a request to join in theory, have you checked? @0x-r4bbit might also been able to help you

**Comment by vkjr** [comment]
Ah, I see! Let me dig deeper)

**Comment by ilmotta** [comment]
@vkjr, do you know if this discussion is still relevant, or better yet, is there a problem we need to solve or has it been solved since December last year? There are some questions you asked that I personally don't know the answer, so maybe they're still relevant and we could ask on the `#communities` channel over Discord.
